android/library/src/main/java/io/logto/android/Logto.kt

```diff
@@ -41,71 +44,98 @@ class Logto(
     val authenticated: Boolean
         get() = tokenSet != null
 
-    fun getAccessToken(block: (accessToken: String) -> Unit) = tokenSet?.let {
-        if (!it.isExpired()) {
-            block(it.accessToken)
-            return@let
+    fun getAccessToken(
+        block: HandleAccessTokenCallback
+    ) {
+        if (tokenSet == null) {
+            block(LogtoException(LogtoException.NOT_AUTHENTICATED), null)
+            return
         }
 
-        refreshTokenSet { newTokenSet ->
-            block(newTokenSet.accessToken)
+        if (!tokenSet!!.isExpired()) {
+            block(null, tokenSet!!.accessToken)
+            return
         }
-    } ?: throw LogtoException(LogtoException.NOT_AUTHENTICATED)
 
-    fun getIdTokenClaims(): JwtClaims {
-        return tokenSet?.getIdTokenClaims()
-            ?: throw LogtoException(LogtoException.NOT_AUTHENTICATED)
+        refreshTokenSet { exception, newTokenSet ->
+            block(exception, newTokenSet?.accessToken)
+        }
+    }
+
+    fun getIdTokenClaims(block: HandleIdTokenClaimsCallback) {
+        if (tokenSet == null) {
+            block(LogtoException(LogtoException.NOT_AUTHENTICATED), null)
+            return
+        }
+        try {
+            val idTokenClaims = tokenSet!!.getIdTokenClaims()
+            block(null, idTokenClaims)
+        } catch (exception: LogtoException) {
+            block(exception, null)
+        }
     }
 
     fun signInWithBrowser(
         context: Context,
-        onComplete: (tokenSet: TokenSet) -> Unit,
-    ) = AuthManager.start(
-        context,
-        BrowserSignInFlow(
+        onComplete: HandleTokenSetCallback,
```

Comment:
just out of curiosity.  when should we name callback attributes as `block` and when other naming